lib/mix/lib/mix/tasks/help.ex

```diff
@@ -86,11 +86,40 @@ defmodule Mix.Tasks.Help do
   def run([task]) do
     loadpaths!()
 
-    {doc, location} = verbose_doc(task)
     opts = Application.get_env(:mix, :colors)
 
+    opts =
+      if ansi_docs?(opts) do
+        [width: width()] ++ opts
+      else
+        opts
+      end
+
+    case verbose_doc(task) do
+      [single] ->
+        print_doc(task, single, opts)
+
+      [alias_doc, task_doc] ->
+        print_doc(task, alias_doc, opts)
+
+        note = "\nThere is also a task named \"#{task}\". The documentation is shown next.\n"
```

Comment:
What if you add this note to `alias_doc` in the line above?

    alias_doc <> "\nThere is also a task named \"#{task}\". The documentation is shown next.\n"

This way you don't need to check for `ansi_docs?(opts)` and the like.